monai/networks/nets/flexible_unet.py

```diff
@@ -0,0 +1,306 @@
+    <https://github.com/qubvel/segmentation_models.pytorch>`_.
+
+    Args:
+        dim: number of spatial dimensions.
+        encoder_channels: number of output channels for all feature maps in encoder.
+            `len(encoder_channels)` should be no less than 2.
+        decoder_channels: number of output channels for all feature maps in decoder.
+            `len(decoder_channels)` should equal to `len(encoder_channels) - 1`.
+        act: activation type and arguments.
+        norm: feature normalization type and arguments.
+        dropout: dropout ratio.
+        upsample: upsampling mode, available options are
+            ``"deconv"``, ``"pixelshuffle"``, ``"nontrainable"``.
+        pre_conv: a conv block applied before upsampling.
+            Only used in the "nontrainable" or "pixelshuffle" mode.
+        interp_mode: {``"nearest"``, ``"linear"``, ``"bilinear"``, ``"bicubic"``, ``"trilinear"``}
+            Only used in the "nontrainable" mode.
+        align_corners: set the align_corners parameter for upsample. Defaults to True.
+            Only used in the "nontrainable" mode.
+
+    """
+
+    def __init__(
+        self,
+        spatial_dims: int,
+        encoder_channels: Sequence[int],
+        decoder_channels: Sequence[int],
+        act: Union[str, tuple],
+        norm: Union[str, tuple],
+        dropout: Union[float, tuple],
+        bias: bool,
+        upsample: str,
+        pre_conv: Optional[str],
+        interp_mode: str,
+        align_corners: Optional[bool],
+    ):
+
+        super().__init__()
+        if len(encoder_channels) < 2:
+            raise ValueError("the length of `encoder_channels` should be no less than 2.")
+        if len(decoder_channels) != len(encoder_channels) - 1:
+            raise ValueError("`len(decoder_channels)` should equal to `len(encoder_channels) - 1`.")
+
+        in_channels = [encoder_channels[-1]] + list(decoder_channels[:-1])
+        skip_channels = list(encoder_channels[1:-1][::-1]) + [0]
+        halves = [True] * (len(skip_channels) - 1)
+        halves.append(False)
+        blocks = []
+        for in_chn, skip_chn, out_chn, halve in zip(in_channels, skip_channels, decoder_channels, halves):
+            blocks.append(
+                UpCat(
+                    spatial_dims=spatial_dims,
+                    in_chns=in_chn,
+                    cat_chns=skip_chn,
+                    out_chns=out_chn,
+                    act=act,
+                    norm=norm,
+                    dropout=dropout,
+                    bias=bias,
+                    upsample=upsample,
+                    pre_conv=pre_conv,
+                    interp_mode=interp_mode,
+                    align_corners=align_corners,
+                    halves=halve,
+                )
+            )
+        self.blocks = nn.ModuleList(blocks)
+
+    def forward(self, features: Sequence[torch.Tensor], skip_connect: int = 4):
+        skips = features[:-1][::-1]
```

Comment:
why not adding the resizing like https://github.com/qubvel/segmentation_models.pytorch/blob/792c2735aa0e926afa17d1105eace63ef2dcc624/segmentation_models_pytorch/decoders/unet/decoder.py#L36
so that we dont have this constraint https://github.com/Project-MONAI/MONAI/blob/e82d15d9c3799b2dbe6a6960cc9e47fb2359e8f7/monai/networks/nets/flexible_unet.py#L297-L300

Reply:
Hi @wyli ,
 I tried to fix this by adding a is_pad parameter in a flexible unet class. Could you please help me to check if this change is ok.